server/src/main/java/org/hl7/davinci/endpoint/files/QuestionnaireValueSetProcessor.java

```diff
@@ -30,11 +30,18 @@ protected Questionnaire processResource(Questionnaire inputResource, FileStore f
     Map<String, ValueSet> valueSetMap = new HashMap<String, ValueSet>();
 
     // Iterate through items recursively and replace answerValueSet appropiately
-    findAndReplaceValueSetReferences(inputResource.getItem(), inputResource.getExtension(), valueSetMap, fileStore, baseUrl);
+    findAndReplaceValueSetReferences(inputResource.getItem(), valueSetMap, fileStore, baseUrl);
 
     // Add all loaded valuesets to the contains field
     for (ValueSet valueSet : valueSetMap.values()) {
       inputResource.addContained(valueSet);
+
+      // START WORKAROUNDS for HAPI Issue: https://github.com/jamesagnew/hapi-fhir/issues/1184
+      // Add initial reference to any item with an answerValueSet due to HAPI encoder error.
```

Comment:
Update this comment to reflect what the new workaround is.

```suggestion
      // Add extension that references the contained ValueSet due to HAPI encoder error.
```